src/qt/diagnosticsdialog.cpp

```diff
@@ -343,102 +607,68 @@ void DiagnosticsDialog::clkFinished()
+    connect(tcpSocket, SIGNAL(error(QAbstractSocket::SocketError)), this, SLOT(TCPFailed(QAbstractSocket::SocketError)));
+
+    tcpSocket->connectToHost("portquiz.net", GetListenPort());
 }
 
-void DiagnosticsDialog::getGithubVersionFinished(QNetworkReply *reply)
+void DiagnosticsDialog::TCPFinished()
 {
-    if (reply->error())
-    {
-        // Incase ssl dlls are not present or corrupted; avoid crash
-        ui->checkClientVersionResultLabel->setText("Failed (" + reply->errorString() + ")");
-
-        return;
-    }
-
-    QByteArray data;
-    data = reply->readAll();
-    std::string newVersionString;
-
-    QJsonDocument replyJson = QJsonDocument::fromJson(data);
-    QJsonObject obj = replyJson.object();
-
-    if (!obj.empty())
-    {
-        QJsonValue newVersionJVal = obj.value("name");
-
-        if (!newVersionJVal.isUndefined())
-            newVersionString = newVersionJVal.toString().toStdString();
-    }
-
-    std::vector<std::string> newVersionStringSplit;
-    boost::algorithm::split(newVersionStringSplit, newVersionString, boost::is_any_of("-"));
-    newVersionString = newVersionStringSplit.at(0);
-
-    std::string currentVersionString = FormatFullVersion();
-    std::vector<std::string> currentVersionStringSplit;
-    boost::algorithm::split(currentVersionStringSplit, currentVersionString, boost::is_any_of("-"));
-    currentVersionString = currentVersionStringSplit.at(0);
-    boost::algorithm::split(currentVersionStringSplit, currentVersionString, boost::is_any_of("v"));
-    currentVersionString = currentVersionStringSplit.at(1);
-
-    std::vector<std::string> currentVersionList;
-    std::vector<std::string> newVersionList;
-
-    boost::algorithm::split(currentVersionList, currentVersionString, boost::is_any_of("."));
-    boost::algorithm::split(newVersionList, newVersionString, boost::is_any_of("."));
+    tcpSocket->close();
+    ui->verifyTCPPortResultLabel->setText(tr("Passed"));
+    ui->verifyTCPPortResultLabel->setStyleSheet("color:black;background-color:green");
 
-    int iNew;
-    int iCurrent;
+    // We need this here because there is no return call (callback) to the on_testButton_clicked function
+    UpdateTestStatus("verifyTCPPort", completed);
+    UpdateOverallDiagnosticResult(passed);
 
-    try
-    {
-        for (unsigned int i=0; i<newVersionList.size(); i++) {
-            iNew = std::stoi(newVersionList.at(i), nullptr, 10);
-            iCurrent = std::stoi(currentVersionList.at(i), nullptr, 10);
+    DisplayOverallDiagnosticResult();
 
-            if (iNew > iCurrent) {
-                ui->checkClientVersionResultLabel->setText("Failed (An update is available, please update to " + QString::fromStdString(newVersionString) + ")");
+    return;
+}
 
-                return;
-            }
+void DiagnosticsDialog::TCPFailed(QAbstractSocket::SocketError socket)
+{
+    ui->verifyTCPPortResultLabel->setText(tr("Failed: Port 32749 may be blocked by your firewall"));
```

Comment:
```suggestion
    ui->verifyTCPPortResultLabel->setText(tr("Warning: Optional full node port 32749 may be blocked by your firewall"));
```
Perhaps this shouldn't be a hard failure. The wallet runs fine without this port open--as you know, it just needs to bind to it to run as a full node.

Reply:
Maybe we should just NA this test unless listen=1 is set.

Reply:
Sorry Jim, I misunderstood this test--it's actually checking for the ability to make an outbound connection to port 32748 and 32749, not for the listening port. 

The original implementation uses the third-party portquiz.net service. This seems fine for now, but we'll need to keep in mind that it may disappear in the future. It may be good to set up a similar service on the official website's domain down the road.

Reply:
Hmm... I will have to add another commit to change it back to not depend on listen=1.